File: src/cs/Samples/Samples/Samples/SolidEdge/Part/SaveWindowAsImageUsingBitBlt.cs
using ApiSamples.Samples.SolidEdge;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ApiSamples.Samples.SolidEdge.Part
{
    /// <summary>
    /// Saves the active 3D window as an image using the BitBlt Win32 function.
    /// </summary>
    class SaveWindowAsImageUsingBitBlt
    {
        static void RunSample(bool breakOnStart)
        {
            if (breakOnStart) System.Diagnostics.Debugger.Break();

            SolidEdgeFramework.Application application = null;
            SolidEdgePart.PartDocument partDocument = null;
            SolidEdgeFramework.Window window = null;

            try
            {
                // Register with OLE to handle concurrency issues on the current thread.
                OleMessageFilter.Register();

                // Connect to Solid Edge.
                application = ApplicationHelper.Connect();

                // Get a reference to the active document.
                partDocument = application.TryActiveDocumentAs<SolidEdgePart.PartDocument>();

                // Make sure we have a document.
                if (partDocument != null)
                {
                    // 3D windows are of type SolidEdgeFramework.Window.
                    window = application.ActiveWindow as SolidEdgeFramework.Window;

                    if (window != null)
                    {
                        WindowHelper.SaveAsImageUsingBitBlt(window);
                    }
                    else
                    {
                        throw new System.Exception(Resources.NoActive3dWindow);
                    }
                }
                else
                {
                    throw new System.Exception(Resources.NoActiveSheetMetalDocument);
                }
            }
            catch (System.Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
            finally
            {
                OleMessageFilter.Unregister();
            }
        }
    }
}

File: src/vb/Samples/ApiSamples/Samples/Part/SaveWindowAsImageUsingBitBlt.vb
Imports SolidEdgeFramework.Extensions 'SolidEdge.Community.dll
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace ApiSamples.Part
	''' <summary>
	''' Saves the active 3D window as an image using the BitBlt Win32 function.
	''' </summary>
	Friend Class SaveWindowAsImageUsingBitBlt
		Private Shared Sub RunSample(ByVal breakOnStart As Boolean)
			If breakOnStart Then
				System.Diagnostics.Debugger.Break()
			End If

			Dim application As SolidEdgeFramework.Application = Nothing
			Dim partDocument As SolidEdgePart.PartDocument = Nothing
			Dim window As SolidEdgeFramework.Window = Nothing

			Try
				' Register with OLE to handle concurrency issues on the current thread.
				SolidEdgeCommunity.OleMessageFilter.Register()

				' Connect to Solid Edge.
				application = SolidEdgeCommunity.SolidEdgeInstall.Start()

				' Get a reference to the active document.
				partDocument = application.GetActiveDocument(Of SolidEdgePart.PartDocument)(False)

				' Make sure we have a document.
				If partDocument IsNot Nothing Then
					' 3D windows are of type SolidEdgeFramework.Window.
					window = TryCast(application.ActiveWindow, SolidEdgeFramework.Window)

					If window IsNot Nothing Then
						WindowHelper.SaveAsImageUsingBitBlt(window)
					Else
						Throw New System.Exception(Resources.NoActive3dWindow)
					End If
				Else
					Throw New System.Exception(Resources.NoActiveSheetMetalDocument)
				End If
			Catch ex As System.Exception
				Console.WriteLine(ex.Message)
			Finally
				SolidEdgeCommunity.OleMessageFilter.Unregister()
			End Try
		End Sub
	End Class
End Namespace
